### PR TITLE
show leader/member information when run 'etcdctl member list' command

### PR DESCRIPTION
**Pull request by mqliang**
This information is very helpful. Otherwise, we must get this information by `curl http://127.0.0.1:2379/v2/stats/self`.

### CODE REVIEW

**Comment by xiang90** [comment]
Probably we should only print out `leader` if that member thinks itself is.

A member can be at `follower`, `candidate` or `leader` state. The state transition is not easy to be understood directly by end user (there might be concurrent leaders, there might be candidate + leader but no follower).

File: etcdctl/command/member_commands.go [hunk 1]
```diff
@@ -63,17 +63,27 @@ func actionMemberList(c *cli.Context) {
...
 	for _, m := range members {
+		status := "member"
+		if m.ID == leader.ID {
+			status = "leader"
+		}
 		if len(m.Name) == 0 {
 			fmt.Printf("%s[unstarted]: peerURLs=%s\n", m.ID, strings.Join(m.PeerURLs, ","))
 		} else {
-			fmt.Printf("%s: name=%s peerURLs=%s clientURLs=%s\n", m.ID, m.Name, strings.Join(m.PeerURLs, ","), strings.Join(m.ClientURLs, ","))
+			fmt.Printf("%s: name=%s peerURLs=%s clientURLs=%s status=%s\n", m.ID, m.Name, strings.Join(m.PeerURLs, ","), strings.Join(m.ClientURLs, ","), status)
```
Comment (xiang90) [inline etcdctl/command/member_commands.go]: isLeader = %v, isLeader

File: etcdctl/command/member_commands.go [hunk 2]
```diff
@@ -63,17 +63,27 @@ func actionMemberList(c *cli.Context) {
...
+	leader, err := mAPI.Leader(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
 
 	for _, m := range members {
+		status := "member"
+		if m.ID == leader.ID {
+			status = "leader"
```
Comment (xiang90) [inline etcdctl/command/member_commands.go]: isLeader = true

File: etcdctl/command/member_commands.go [hunk 3]
```diff
@@ -63,17 +63,27 @@ func actionMemberList(c *cli.Context) {
...
 		os.Exit(1)
 	}
+	leader, err := mAPI.Leader(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
 
 	for _, m := range members {
+		status := "member"
```
Comment (xiang90) [inline etcdctl/command/member_commands.go]: isLeader := false

File: etcdctl/command/member_commands.go [hunk 4]
```diff
@@ -63,17 +63,27 @@ func actionMemberList(c *cli.Context) {
...
 	defer cancel()
+
 	members, err := mAPI.List(ctx)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err.Error())
 		os.Exit(1)
 	}
+	leader, err := mAPI.Leader(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
```
Comment (xiang90) [inline etcdctl/command/member_commands.go]: os.Stderr, "Failed to get leader %v", err

**Comment by mqliang** [comment]
@xiang90 Thanks for your patient explanation. Updated, PTAL.

**Comment by xiang90** [comment]
gofmt fails. please fix.

**Comment by xiang90** [comment]
LGTM. Please format the commit message to etcdctl: print out leader info when list member. (please follow https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message)

**Comment by xiang90** [comment]
Thanks. will merge after CI turns green.

**Comment by xiang90** [comment]
@mqliang We plan to use "github.com/olekukonko/tablewriter" to make etcdctlv3 member list more pretty.  You can take a look if you would like to. : ) Thanks for contribution!

**Comment by mqliang** [comment]
@xiang90 
> We plan to use "github.com/olekukonko/tablewriter" to make etcdctlv3 member list more pretty. You can take a look if you would like to

Thanks. I will take a look.

BTW, I'd like add a health check route in https://github.com/coreos/etcd/blob/master/client/client.go#L384. If client pins to a member and unfortunately this member isolated for cluster, PUT/POST request will failed, whereas Get request will succeeded but return stale data. 

If client unfortunately just send Get requests for a period of time, it can NOT aware it get stale data. Enable quorum read could solve problem this but it make Get request less efficient

Not sure if we need do this. Expecting your suggestion.

**Pull request merged**